src/Monolog/Handler/StreamHandler.php

```diff
@@ -50,9 +54,26 @@ class StreamHandler extends AbstractProcessingHandler
     public function __construct($stream, $level = Logger::DEBUG, bool $bubble = true, ?int $filePermission = null, bool $useLocking = false)
     {
         parent::__construct($level, $bubble);
+
+        if ($phpMemoryLimit = ini_get('memory_limit')) {
+            if (($memoryInByes = Utils::memoryIniValueToBytes($phpMemoryLimit))) {
+                $memoryUsage = memory_get_usage(true);
+                if (($memoryInByes - $memoryUsage) < $this->streamChunkSize) {
+                    $this->streamChunkSize = $memoryInByes - $memoryUsage - self::SAFE_MEMORY_OFFSET;
+                }
+            }
+        }
+
         if (is_resource($stream)) {
             $this->stream = $stream;
-            stream_set_chunk_size($this->stream, self::MAX_CHUNK_SIZE);
+
+            try {
+                stream_set_chunk_size($this->stream, $this->streamChunkSize);
+            } catch (\Exception $exception) {
+                throw new \RuntimeException('Impossible to set the stream chunk size.'
+                .PHP_EOL.'Error: '.$exception->getMessage()
+                .PHP_EOL.'Trace: '.$exception->getTraceAsString());
+            }
```

Comment:
Why rethrowing here and not simply letting the original exception do its thing?

Reply:
I'll revert this change for now as I don't understand it and I'd like to get the rest merged, if you think it's important please explain and we can do it in another PR.